### ISSUE TITLE
Navigation Compose Multiplatform 0.0.0-nav-dev1535 - java.lang.IllegalArgumentException when navigate with arguments

### ISSUE THREAD

**Issue by serbelga** [JetBrains/compose-multiplatform#4633]
I was trying to use the new Navigation Compose Multiplatform implementation on my project https://github.com/serbelga/Todometer-KMP/pull/515 and there's a crash on iOS and Desktop platforms when navigate with arguments.

Desktop:

```
Caused by: java.lang.IllegalArgumentException: Navigation destination that matches route taskdetail/d3004435-4d3d-45bd-905b-67429eea7668 
cannot be found in the navigation graph ComposeNavGraph startDestination={Destination route=home}
```  

**Affected platforms**

- Desktop (Windows, Linux, macOS)
- iOS

**Versions**
- Kotlin version: 1.9.23
- Compose Multiplatform version: 1.6.10-dev1578
- OS version(s): Mac OS 14.1.1 (23B81)
- OS architecture: arm64
- JDK: 17

**Screenshots**

Here's a screenshot of debugger on `composable` function in NavHost

<img width="640" alt="Captura de pantalla 2024-04-16 a las 20 13 20" src="https://github.com/JetBrains/compose-multiplatform/assets/26246782/e3eeb275-ecf3-4310-a566-24e79888b3af">

I'm defining this route as "taskdetail/{taskId}" as AndroidX Navigation Compose documentation specifies.

**Comment by PMARZV** [comment]
Its been fixed in this pull request theoretically (merged today) https://github.com/JetBrains/compose-multiplatform-core/pull/1277

**Comment by MatkovIvan** [comment]
Yes, it's fixed by https://github.com/JetBrains/compose-multiplatform-core/pull/1277

It will be available in the next build

**Comment by serbelga** [comment]
Thanks!! 😉

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4633">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
